src/coreComponents/linearAlgebra/DofManager.cpp

```diff
@@ -905,11 +910,11 @@ void DofManager::setSparsityPatternFromStencil( SparsityPatternView< globalIndex
     forMeshLocation< FieldLocation::Elem, false, parallelHostPolicy >( mesh, regions, [=]( auto const & elemIdx )
     {
       globalIndex const elemDof = dofNumberView[elemIdx[0]][elemIdx[1]][elemIdx[2]];
-      for( localIndex c = 0; c < numComp; ++c )
+      for( integer c = 0; c < numComp; ++c )
       {
         colDofIndices[c] = elemDof + c;
       }
-      for( localIndex c = 0; c < numComp; ++c )
+      for( integer c = 0; c < numComp; ++c )
       {
         pattern.insertNonZeros( elemDof - rankDofOffset + c, colDofIndices.begin(), colDofIndices.end() );
```

Comment:
`elemDof` and `rankDofOffset` are of type `globalIndex`, while `c` is `integer`. We should fix the type mismatch

Reply:
My logic behind these type choices:
  * `globalIndex` - anything that enumerates things (elems, nodes, etc.) in the global mesh
  * `localIndex` - same but in a local mesh partition
  * `integer` - all other integers that don't "scale" with the size of the mesh. In particular, while `localIndex` can be 64-bit in some builds, `integer` is always 32-bit, so it's more suitable to represent small quantities.

By this logic, "small" numbers like number of DoF components in a field or component index should be of type `integer`. They can be freely added to `globalIndex` variables to form new global DoF indices - there is no need to change types here IMO.

Reply:
Makes sense, thanks!